main.go

```diff
@@ -56,40 +59,52 @@ var (
 	deferred []func() error
 	fset     = token.NewFileSet()
 
-	b64             = base64.NewEncoding("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_z")
-	printerConfig   = printer.Config{Mode: printer.RawFormat}
-	origTypesConfig = types.Config{Importer: importer.ForCompiler(fset, "gc", origLookup)}
+	b64           = base64.NewEncoding("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_z")
+	printerConfig = printer.Config{Mode: printer.RawFormat}
+
+	// listPackage helps implement a types.Importer which finds the export
+	// data for the original dependencies, not their garbled counterparts.
+	// This is useful to typecheck a package before it's garbled, so we can
+	// make decisions on how to garble it.
+	origTypesConfig = types.Config{Importer: importer.ForCompiler(fset, "gc", func(path string) (io.ReadCloser, error) {
+		pkg, err := listPackage(path)
+		if err != nil {
+			return nil, err
+		}
+		return os.Open(pkg.Export)
+	})}
 
 	buildInfo       = packageInfo{imports: make(map[string]importedPkg)}
 	garbledImporter = importer.ForCompiler(fset, "gc", func(path string) (io.ReadCloser, error) {
 		return os.Open(buildInfo.imports[path].packagefile)
 	}).(types.ImporterFrom)
 
-	envGarbleDir = os.Getenv("GARBLE_DIR")
-	envGoPrivate string // filled via 'go env' below to support 'go env -w'
+	envGarbleDir  = os.Getenv("GARBLE_DIR")
+	envObfStrings = os.Getenv("OBF_STRINGS") == "true"
```

Comment:
This should have a `GARBLE_` prefix, otherwise we risk clashing with environment variables from other programs.

Also, replace "strings" with "literals", just like the flag.

Reply:
Done